pkg/cmd/extension/manager.go

```diff
@@ -548,23 +569,27 @@ func (m *Manager) upgradeExtension(ext Extension, force bool) error {
 			isBin, _ = isBinExtension(m.client, repo)
 		}
 		if isBin {
-			err = m.Remove(ext.Name())
-			if err != nil {
-				return fmt.Errorf("failed to migrate to new precompiled extension format: %w", err)
+			if !dryRun {
+				if err := m.Remove(ext.Name()); err != nil {
```

Comment:
When `dryRun` is a property of Manager, then instead of performing a conditional, this can still invoke `Remove()`, but the function could be a no-op.